server/routes/admin/onscreen_limits.py

```
@@ -0,0 +1,46 @@
+    validated, errors = validate_request(OnscreenLimitsSchema, payload)
+    if errors:
+        return _json_response({"error": "Validation failed", "details": errors}, 400)
+
+    try:
+        new_state = onscreen_config.set_state(
+            max_onscreen_danmu=validated["max_onscreen_danmu"],
+            overflow_mode=validated["overflow_mode"],
+        )
+    except ValueError as exc:
+        return _json_response({"error": str(exc)}, 400)
+    except Exception as exc:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)